### PR TITLE
feat: migrate from "aioredis" to "redis" with keeping optional "aioredis" support

### PR DESCRIPTION
**Pull request by AIGeneratedUsername**
"aioredis" now is a part of "redis", so no need to use "aioredis" anymore. See [this note](https://github.com/aio-libs/aioredis-py#-aioredis-is-now-in-redis-py-420rc1-)

"aioredis" still can be installed if someone needs it

---

Solves https://github.com/Krukov/cashews/issues/47

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -24,14 +25,14 @@ scalable and reliable applications. This library intends to make it easy to impl
 - Easy to configure and use
 - Decorator-based API, just decorate and play
 - Different cache strategies out-of-the-box
-- Support for multiple storage backends ([In-memory](#in-memory), [Redis](#redis), [DiskCache](diskcache))
```
Comment (AIGeneratedUsername) [inline Readme.md:27]: > [In-memory](#in-memory)

Is there any hidden purposes behind this syntax? As I know, it is only for URLs.\
I think using **bold**, or _italic_, or "quotes" is enough.

File: cashews/backends/redis/compat.py [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
-    from aioredis import BlockingConnectionPool
-    from aioredis import ConnectionError as RedisConnectionError
-    from aioredis import Redis
+    from redis.asyncio import BlockingConnectionPool, Redis
+    from redis.exceptions import ConnectionError as RedisConnectionError
 
-    _AIOREDIS = "2"
+    _AIOREDIS = "2"  # Technically, not "aioredis" anymore (just "redis"). Keep this name as long as we support "aioredis".
 
 except ImportError:
```
Comment (Krukov) [inline cashews/backends/redis/compat.py:8]: I assume a little bit more changes here - let's drop support aioredis version 1.x.x support and remove all code for supporting it. 
Block bellow with ImportError should contain :
```python
    from aioredis import BlockingConnectionPool
    from aioredis import ConnectionError as RedisConnectionError
    from aioredis import Redis
```

**Comment by AIGeneratedUsername** [comment]
I do not know why, but one test failed for previous tests run: https://github.com/Krukov/cashews/runs/6442980803?check_suite_focus=true

File: cashews/backends/redis/compat.py (hunk 2 above)
Comment (AIGeneratedUsername) [reply cashews/backends/redis/compat.py]: aioredis 1.0 support dropped

File: cashews/backends/client_side.py [hunk 3]
```diff
@@ -27,11 +27,10 @@
 import asyncio
 import logging
 
+from redis.exceptions import ConnectionError as RedisConnectionError
```
Comment (Krukov) [inline cashews/backends/client_side.py:30]: Please add here logic for import error  to be compatible with aioredis

File: setup.cfg [hunk 4]
```diff
@@ -31,6 +31,8 @@ zip_safe = false
 
 [options.extras_require]
 redis = 
+    redis >= 4.3.1
+aioredis =
     aioredis < 3.0.0
```
Comment (Krukov) [inline setup.cfg:36]: can you also restrict here aioredis <= 2.0.0 ( I don't know how to do it exactly)

**Comment by Krukov** [comment]
Great job, thank you so much

**Pull request merged**